internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_controller.go

```diff
@@ -316,6 +352,25 @@ func (c *Controller) runPhase(ctx context.Context, phase runtime.Phase, seq runt
 
 	var eg errgroup.Group
 
+	doneCh := make(chan struct{})
+
+	eg.Go(func() error {
+		done := 0
+
+		for {
```

Comment:
why do we need this?

it feels like it's re-doing what `errgroup` should already be doing

Reply:
to cancel waiting for errgroup if the context is canceled and any of the underlying task doesn't cancel when the context is canceled.

Reply:
if the task doesn't cancel, we can't cancel either way, as the task is still running?

we should in fact use https://pkg.go.dev/golang.org/x/sync/errgroup#WithContext, as it automatically cancels the context when any of the sub-goroutines fails.

Reply:
oh, I didn't know that this thing exists. Will update